client/src/components/LinkField/LinkField.js

```diff
@@ -27,24 +26,29 @@ const LinkField = ({id, loading, Loading, data, LinkPicker, onChange, types, lin
 
   const linkProps = {
     title: data ? data.Title : '',
-    link: type ? {type, title: data.Title, description: linkDescription} : undefined,
-    onEdit: () => {setEditing(true)},
+    link: type ? { type, title: data.Title, description: linkDescription } : undefined,
+    onEdit: () => { setEditing(true); },
     onClear,
     onSelect: (key) => {
       setNewTypeKey(key);
       setEditing(true);
     },
     types: Object.values(types)
-  }
+  };
 
-  const onModalSubmit = (data, action, submitFn) => {
-    console.dir({data, action, submitFn, onChange});
-    const {SecurityID, action_insert, ...value} = data;
-    typeof onChange === 'function' && onChange(event, { id, value})
+  const onModalSubmit = (modalData, action, submitFn) => {
+    const { SecurityID, action_insert: actionInsert, ...value } = modalData;
+
+    if (typeof onChange !== 'function') {
+      return Promise.resolve();
+    }
+
+    onChange(event, { id, value });
     setEditing(false);
     setNewTypeKey('');
```

Comment:
```suggestion
    if (typeof onChange === 'function') {
      onChange(event, { id, value });
      setEditing(false);
      setNewTypeKey('');
    }
```
This is subjective though so take it or leave it

Reply:
Also... are these lines even related to whether we have an onChange callback or not?
```js
setEditing(false);
setNewTypeKey('');
```

They feel separate.

Reply:
yeah good point - was mainly fixing a linting error will move these up